mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -140,7 +143,10 @@ class LinearRegression(override val uid: String)
       logWarning(s"The standard deviation of the label is zero, so the weights will be zeros " +
         s"and the intercept will be the mean of the label; as a result, training is not needed.")
       if (handlePersistence) instances.unpersist()
-      return new LinearRegressionModel(uid, Vectors.sparse(numFeatures, Seq()), yMean)
+      val weights = Vectors.sparse(numFeatures, Seq())
+      val intercept = yMean
+      val summary = generateTrainingResults(instances, Array(), weights, intercept)
```

Comment:
Nope. If all labels are the same there is no objectiveTrace and all the weights are zero (since just predicting `yMean` = the single correct label will give 0 loss).